src/System.Security.Cryptography.Pkcs/tests/SignedCms/SignedCmsTests.cs

```diff
@@ -933,5 +933,173 @@ public static void SignerInfoCollection_Indexer_MinusOne ()
+        public static void SignIdentifiedContent(string oidValue, string contentHex, bool netfxProblem)
+        {
+            SignedCms signedCms = new SignedCms(
+                new ContentInfo(new Oid(oidValue, "Some Friendly Name"), contentHex.HexToByteArray()));
+
+            using (X509Certificate2 cert = Certificates.RSAKeyTransferCapi1.TryGetCertificateWithPrivateKey())
+            {
+                try
+                {
+                    signedCms.ComputeSignature(new CmsSigner(cert));
+                }
+                catch (CryptographicException) when (netfxProblem)
+                {
+                    // When no signed or unsigned attributes are present and the signer uses
+                    // IssuerAndSerial as the identifier type, NetFx uses an older PKCS7 encoding
+                    // of the current CMS one.  The older encoding fails on these inputs because of a
+                    // difference in PKCS7 vs CMS encoding of values using types other than Pkcs7Data.
+                    return;
+                }
+
+                byte[] encoded = signedCms.Encode();
+                signedCms.Decode(encoded);
+            }
+
+            // Assert.NoThrows
+            signedCms.CheckSignature(true);
+
+            Assert.Equal(oidValue, signedCms.ContentInfo.ContentType.Value);
+            Assert.Equal(contentHex, signedCms.ContentInfo.Content.ByteArrayToHex());
+        }
+
+        [Theory]
+        [InlineData(null, "0102", Oids.Pkcs7Data)]
+        [InlineData(null, "010100", Oids.Pkcs7Data)]
+        [InlineData("potato", "010100", null)]
+        [InlineData(" 1.1", "010100", null)]
+        [InlineData("1.1 ", "010100", null)]
+        [InlineData("1 1", "010100", null)]
+        public static void SignIdentifiedContent_BadOid(string oidValueIn, string contentHex, string oidValueOut)
+        {
+            SignedCms signedCms = new SignedCms(
+                new ContentInfo(new Oid(oidValueIn, "Some Friendly Name"), contentHex.HexToByteArray()));
+
+            using (X509Certificate2 cert = Certificates.RSAKeyTransferCapi1.TryGetCertificateWithPrivateKey())
+            {
+                Action signAction = () => signedCms.ComputeSignature(new CmsSigner(cert));
+
+                if (oidValueOut == null)
+                {
+                    Assert.ThrowsAny<CryptographicException>(signAction);
+                    return;
+                }
+
+                signAction();
+
+                byte[] encoded = signedCms.Encode();
+                signedCms.Decode(encoded);
+            }
+
+            // Assert.NoThrows
+            signedCms.CheckSignature(true);
+
+            Assert.Equal(oidValueOut, signedCms.ContentInfo.ContentType.Value);
+            Assert.Equal(contentHex, signedCms.ContentInfo.Content.ByteArrayToHex());
+        }
+
+        [Fact]
+        public static void CheckSignedEncrypted_IssuerSerial_FromNetFx()
+        {
+            CheckSignedEncrypted(
```

Comment:
Those can be collapsed into a single theory with MemberData - I'm fine either way though

Reply:
Yep, but when passing in `byte[]` that produces test names that are hard to understand in failure messages and result logs.